pkg/frontegg/app_password.go

```diff
@@ -0,0 +1,71 @@
+package frontegg
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/clients"
+)
+
+type AppPasswordResponse struct {
```

Comment:
With the SQL builder, we have a separate package called `materialize` that is purely responsible for building the SQL statements. I am planning to also do the same for the Frontegg API, that way the `frontegg` package will hold all of the logic for interacting with the Frontegg API and then those funcs will be used across the different packages like data sources, resources and tests.

Initially this all lived under the `resources` package but with the additional resources that we've added recently the resource logic is getting too noisy.